plugins/simpleadminhacks-paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/CaseInsensitivity.java

```diff
@@ -0,0 +1,24 @@
+package com.programmerdan.minecraft.simpleadminhacks.hacks.basic;
+
+import com.programmerdan.minecraft.simpleadminhacks.SimpleAdminHacks;
+import com.programmerdan.minecraft.simpleadminhacks.framework.BasicHack;
+import com.programmerdan.minecraft.simpleadminhacks.framework.BasicHackConfig;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.player.PlayerCommandPreprocessEvent;
+
+public class CaseInsensitivity extends BasicHack {
+
+    public CaseInsensitivity(SimpleAdminHacks plugin, BasicHackConfig config) {
+        super(plugin, config);
+    }
+
+    @EventHandler
```

Comment:
I'd argue that the priority should be set to LOWEST so that the transformation is done as early as possible. Any event handlers thereafter would then be deciding based on the amended command.